engine/v2/transaction/record/custom_outputs.go

```diff
@@ -0,0 +1,91 @@
+package record
+
+import (
+	"iter"
+
+	primitives "github.com/bitcoin-sv/go-sdk/primitives/ec"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/custominstructions"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction"
+	"github.com/bitcoin-sv/spv-wallet/engine/v2/transaction/txmodels"
+	"github.com/bitcoin-sv/spv-wallet/models/bsv"
+	"github.com/bitcoin-sv/spv-wallet/models/transaction/bucket"
+)
+
+type customOutputsResolver struct {
+	flow        *txFlow
+	userID      string
+	userPubKey  *primitives.PublicKey
+	annotations transaction.OutputsAnnotations
+	txAddresses addresses
+	err         error
+}
+
+func (c *customOutputsResolver) getUserPubKey() (*primitives.PublicKey, error) {
+	if c.userPubKey == nil {
+		pubKey, err := c.flow.service.users.GetPubKey(c.flow.ctx, c.userID)
+		if err != nil {
+			return nil, spverrors.Wrapf(err, "failed to get public key for user %s", c.userID)
+		}
+		c.userPubKey = pubKey
+	}
+	return c.userPubKey, nil
+}
+
+func (c *customOutputsResolver) remainingAddresses() addresses {
+	return c.txAddresses
+}
+
+func (c *customOutputsResolver) resolveAddress(address string, vout uint32) {
+	c.txAddresses.remove(address, vout)
+}
+
+func (c *customOutputsResolver) annotatedOutputs() iter.Seq2[string, txmodels.NewOutput] {
+	return func(yield func(address string, output txmodels.NewOutput) bool) {
+		for vout, annotation := range c.annotations {
+			if annotation.Bucket != bucket.BSV || annotation.CustomInstructions == nil {
+				continue
+			}
+
+			userPubKey, err := c.getUserPubKey()
+			if err != nil {
+				c.err = spverrors.Wrapf(err, "failed to get public key for user %s", c.userID)
+				break
+			}
+
+			interpreted, err := custominstructions.NewAddressInterpreter().
+				Process(userPubKey, *annotation.CustomInstructions)
+			if err != nil {
+				c.err = spverrors.Wrapf(err, "failed to derive address from custom instructions for user %s", c.userID)
+				break
+			}
+
+			_, ok := c.txAddresses[interpreted.Address.AddressString]
```

Comment:
I totally don't understand, why we're storing all addresses and why we're looking for them.

I mean: we've got annotation that is strictly saying:
> Output X: will be unlocked with this custom instructions
 
But then, we're checking if those custom instructions will unlock any of Outputs, not only X

Reply:
Ok, despite this address extaction,
why are we doing double check
first if there is this address at all
second if it is in provided vout

IMHO we should check only if it is in that particular output, and that's all.